pkg/token/issuertoken/issuertoken_test.go

```diff
@@ -0,0 +1,49 @@
+// (C) Copyright 2024 Hewlett Packard Enterprise Development LP
+
+package issuertoken
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/hewlettpackard/hpegl-provider-lib/pkg/provider"
+)
+
+func TestGenerateParamsAndURL(t *testing.T) {
+	t.Parallel()
+	testcases := []struct {
+		name       string
+		iamVersion string
+		hasError   bool
+	}{
+		{
+			name:       "valid IAM version GLCS",
+			iamVersion: provider.IAMVersionGLCS,
+			hasError:   false,
+		},
+		{
+			name:       "valid IAM version GLP",
+			iamVersion: provider.IAMVersionGLP,
+			hasError:   false,
+		},
+		{
+			name:       "invalid IAM version",
+			iamVersion: "invalid",
+			hasError:   true,
+		},
+	}
+
+	for _, testcase := range testcases {
+		tc := testcase
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			_, _, err := generateParamsAndURL("clientID", "clientSecret", "identityServiceURL", tc.iamVersion)
```

Comment:
We should probably also verify that at least the returned URL is correct. Might be good to check that the params are all good too.

Reply:
I've added a validation function on the value of `iam_service_url`.  Also checking params here.